apis/v1alpha1/gateway_types.go

```diff
@@ -717,38 +717,140 @@ type ListenerStatus struct {
+	ListenerConditionConflicted ListenerConditionType = "Conflicted"
+
+	// ListenerReasonHostnameConflict is used when the Listener
+	// violates the Hostname match constraints for that allow
+	// collapsing Listeners. For example, this reason would be
+	// used when multiple Listeners on the same port use the
+	// "Any" hostname match type.
+	ListenerReasonHostnameConflict ListenerConditionReason = "HostnameConflict"
+
+	// ListenerReasonProtocolConflict is used when multiple
+	// Listeners are specified with the same Listener port number,
+	// but have conflicting protocol specifications.
+	ListenerReasonProtocolConflict ListenerConditionReason = "ProtocolConflict"
+
+	// ListenerReasonRouteConflict is used when the route
+	// resources selected for this Listener conflict with other
+	// specified properties of the Listener (e.g. Protocol).
+	// For example, a Listener that specifies "UDP" as the protocol
+	// but a route selector that resolves "TCPRoute" objects.
+	ListenerReasonRouteConflict ListenerConditionReason = "RouteConflict"
+)
+
+const (
+	// ListenerConditionDetached indicates that, even though
+	// the listener is syntactically and semantically valid, the
+	// controller is not able to configure it on the underlying
+	// Gateway infrastructure.
+	//
+	// A Listener is specified as a logical requirement, but needs to be
+	// configured on a network endpoint (i.e. address and port) by a
+	// controller. The controller may be unable to attach the Listener
+	// if it specifies an unsupported requirement, or prerequisite
+	// resources are not available.
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "PortUnavailable"
+	// * "UnsupportedExtension"
+	// * "UnsupportedProtocol"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ListenerConditionDetached ListenerConditionType = "Detached"
+
+	// ListenerReasonPortUnavailable is used when the Listener
+	// requests a port that cannot be used on the Gateway.
+	ListenerReasonPortUnavailable ListenerConditionReason = "PortUnavailable"
+
+	// ListenerReasonUnsupportedExtension is used when the
+	// controller detects that an implementation-specific Listener
+	// extension is being requested, but is not able to support
+	// the extension.
+	ListenerReasonUnsupportedExtension ListenerConditionReason = "UnsupportedExtension"
+
+	// ListenerReasonUnsupportedProtocol is used when the
+	// Listener could not be attached to be Gateway because its
+	// protocol type is not supported.
+	ListenerReasonUnsupportedProtocol ListenerConditionReason = "UnsupportedProtocol"
+)
+
+const (
+	// ListenerConditionResolvedRefs indicates whether the
+	// controller was able to resolve all the object references
+	// for the Listener.
+	//
+	// Possible reasons for this condition to be false are:
+	//
+	// * "DroppedRoutes"
+	// * "InvalidCertificateRef"
+	// * "InvalidRoutesRef"
```

Comment:
Maybe we should provide some guidance as far as the order of when these should be applied if they are all true? (Assuming only one entry for this condition will exist at a given time).

Reply:
> Assuming only one entry for this condition will exist at a given time

Yes, Condition semantics are that each type is always present, exactly once.